lib/index.js

```diff
@@ -104,45 +117,78 @@ exports.parse = function (definition, options) {
 
             let value = arg;
             if (last) {
-                if (last.type === 'number') {
+                if (last.def.type === 'number') {
                     value = parseInt(arg, 10);
 
                     if (!Number.isSafeInteger(value)) {
-                        errors.push(internals.formatError('Invalid value (non-number) for option:', last.name));
+                        errors.push(internals.formatError('Invalid value (non-number) for option:', last.def.name));
                         continue;
                     }
                 }
 
-                if (last.valid &&
-                    last.valid.indexOf(value) === -1) {
+                if (last.def.valid &&
+                    !last.def.valid.includes(value)) {
 
                     const validValues = [];
-                    for (let j = 0; j < last.valid.length; ++j) {
-                        const valid = last.valid[j];
+                    for (let j = 0; j < last.def.valid.length; ++j) {
+                        const valid = last.def.valid[j];
                         validValues.push(`'${valid}'`);
                     }
 
-                    errors.push(internals.formatError('Invalid value for option:', last.name, '(valid: ' + validValues.join(',') + ')'));
+                    errors.push(internals.formatError('Invalid value for option:', last.def.name, '(valid: ' + validValues.join(',') + ')'));
                     continue;
                 }
+
+                if (last.def.type === 'json') {
+
+                    const stringValue = value;
+
+                    try {
+                        value = Bourne.parse(value, {   // E.g. { x }
+                            protoAction: 'remove'
+                        });
+                    }
+                    catch (err) {
+                        // If the input doesn't look like JSON, we'll ignore that and treat it as a string
+                        Bounce.ignore(err, SyntaxError);
+                    }
+
+                    if (!last.def.parsePrimitives && (!value || typeof value !== 'object')) {
```

Comment:
Why do you want to always parse objects & arrays? It means that even with `parsePrimitives: false`, someone that expects a simple string could crash the app with `{"toString":null}`, possibly producing unexpected results.

Reply:
If someone is looking to guarantee they receive a string input rather than json input then they would want to use a dot-separated `string`-type arg, then optionally use `Bossy.object()` to roll it up into an object.  This can even be combined with a `json`-type arg to get the best of both worlds and sidestep any unpredictability if the application would like to guarantee certain deep properties be strings.